grobid-core/src/main/java/org/grobid/core/document/TEIFormatter.java

```diff
@@ -747,6 +747,10 @@ else if (biblio.getE_Year().length() == 4)
             tei.append("\t\t\t\t\t<idno type=\"DOI\">" + TextUtilities.HTMLEncode(theDOI) + "</idno>\n");
         }
 
+        if (!StringUtils.isEmpty(biblio.getHalId())) {
```

Comment:
I think here we can use StringUtils.isNotBlank() also to avoid having empty strings